### PR TITLE
Update localization docs for v3.3

### PR DESCRIPTION
**Pull request by jasonblais**

### CODE REVIEW

**Comment by esethna** [comment]
+1, thanks Jason, please make sure you verify the rendering in Sphinx once it is built on docs

File: source/developer/localization-process.rst [hunk 1]
```diff
@@ -0,0 +1,116 @@
+Localization
+------------------
+
+The goal of the localization process is to consistently produce high quality translations release after release through collaboration between the international community, translators, project managers, language leads and the core team.
+
+Translation Process
+===================
+
+If you're interested in contributing translations to Mattermost, please join the `Mattermost localization channel to discuss <https://pre-release.mattermost.com/core/channels/localization>`_. Below are descriptions of the overall localization process.
```
Comment (lindy65) [inline source/developer/localization-process.rst]: Seems the `_` before the fullstop at the end of the first sentence shouldn't be there?

File: source/developer/localization-process.rst [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+Translation Quality
+===================
+
+So users understand the accuracy and coverage of language translations, quality levels are explicitly defined for each language:
+
+-----------------
+Official
+-----------------
+- 100% of translation verified by someone deeply knowledgeble in Mattermost functionality.
```
Comment (lindy65) [inline source/developer/localization-process.rst]: Amend throughout doc:
`knowledgeble` > `knowledgeable`

File: source/developer/localization-process.rst [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+===================
+
+Translations require updates on a monthly basis as features are added and changed. The formal process for updates has yet to be determined.
+
+Below are current official reviewers and maintainers for languages that have reached at least Beta-quality. Official reviewers submit final translations for languages; maintainers suggest translations and step in when official reviewers aren't able to help in a certain release. 
+
+If you're interested in contributing to the process, please join the `Mattermost localization channel to discuss <https://pre-release.mattermost.com/core/channels/localization>`_.
+
++------------------------------------------+-----------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------+
+| Langugage                                | Official Reviewer(s)                                                                                                  | Maintainers                                                                                          |
```
Comment (lindy65) [inline source/developer/localization-process.rst]: `Langugage` > `Language`

**Comment by jasonblais** [comment]
Thanks, typos fixed @lindy65

**Pull request merged**